src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -1324,35 +1324,15 @@ private DebugId GetLambdaId(SyntaxNode syntax, ClosureKind closureKind, int clos
             if (_analysis.LambdaScopes.TryGetValue(node.Symbol, out lambdaScope))
             {
                 containerAsFrame = _frames[lambdaScope];
-                var structClosureParamBuilder = ArrayBuilder<TypeSymbol>.GetInstance();
-                while (containerAsFrame != null && containerAsFrame.IsValueType)
+                structClosures = _analysis.MethodsConvertedToDelegates.Contains(node.Symbol)
```

Comment:
Would there be a problem with async/iterator methods?

---
In reply to: [100203934](https://github.com/dotnet/roslyn/pull/16908#discussion_r100203934) [](ancestors = 100203934)